pkg/internal_registry/registry.go

```diff
@@ -0,0 +1,57 @@
+package registry
+
+import (
+	brokerModel "github.com/eclipse/che-plugin-broker/model"
+)
+
+func InternalRegistry() map[string]brokerModel.PluginMeta {
+
+	return map[string]brokerModel.PluginMeta{
+		"eclipse/cloud-shell/nightly": {
+			APIVersion:  "v2",
+			Publisher:   "eclipse",
+			Description: "Cloud Shell provides an ability to use terminal widget like an editor.",
+			DisplayName: "Cloud Shell Editor",
+			ID:          "",
+			Icon:        "https://www.eclipse.org/che/images/logo-eclipseche.svg",
+			Name:        "cloud-shell",
+			Spec: brokerModel.PluginMetaSpec{
+				Endpoints: []brokerModel.Endpoint{
+					{
+						Name:       "cloud-shell",
+						Public:     true,
+						TargetPort: 4444,
+						Attributes: map[string]string{
+							"protocol":           "http",
+							"type":               "ide",
+							"discoverable":       "false",
+							"secure":             "true",
+							"cookiesAuthEnabled": "true",
+						},
+					},
+				},
+				Containers: []brokerModel.Container{
+					{
+						Name:  "che-machine-exec",
+						Image: "quay.io/eclipse/che-machine-exec:nightly",
+						Ports: []brokerModel.ExposedPort{
+							{
+								ExposedPort: 4444,
+							},
+						},
+						Command: []string{
+							"/go/bin/che-machine-exec",
+							"--static",
+							"/cloud-shell",
+							"--url",
+							"127.0.0.1:4444",
```

Comment:
I put a `json.MarshalIndent` on the plugin meta we get from `utils.GetPluginMeta` and I see (formatted as json)
```yaml
apiVersion: v2
spec:
  endpoints:
  - name: cloud-shell
    public: true
    targetPort: 4444
    attributes:
      cookiesAuthEnabled: "true"
      discoverable: "false"
      protocol: http
      secure: "true"
      type: ide
  containers:
  - name: che-machine-exec
    image: quay.io/eclipse/che-machine-exec:nightly
    env: []
    commands: []
    volumes: []
    ports:
    - exposedPort: 4444
    mountSources: false
    command: []
    args: []
  initContainers: []
  workspaceEnv: []
  extensions: []
id: eclipse/cloud-shell/nightly
name: cloud-shell
displayName: Cloud Shell Editor
publisher: eclipse
type: Che Editor
description: Cloud Shell provides an ability to use terminal widget like an editor.
version: nightly
title: Cloud Shell Editor
icon: https://www.eclipse.org/che/images/logo-eclipseche.svg

Reply:
The command in your meta.yaml was added in https://github.com/eclipse/che-plugin-registry/commit/74fcb49, which is not included in the 7.8.0 release of the registry. I don't know if it matters significantly, but this does bring up the issue of how we maintain the offline registry (we can discuss in a separate issue)

Reply:
The command actually prevents machine-exec from listening outside the local network, which means cloud shell won't work with basic routing. It's not an issue for openshift-oauth as the proxy pod is included in the deployment and forwards requests.

Created issue https://github.com/eclipse/che/issues/16568 to discuss this.

Reply:
Probably we also may need some endpoint diagnostics tools in workspace controller, like now with nightly plugin registry we'll get Running but not working workspace
![Screenshot_20200408_124846](https://user-images.githubusercontent.com/5887312/78770608-777e2600-7997-11ea-978f-4c95330dbfd5.png)